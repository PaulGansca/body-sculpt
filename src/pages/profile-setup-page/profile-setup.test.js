import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react'
import ProfileSetupPage from './profile-setup-page';
import ProfileSetupPageContainer from './profile-setup-page.container';

let wrapper;

beforeEach(() => {
    const mockProps = {
        currentUser: {id: "bla123bla", goalSet: true, birthday: new Date()},
        loading: false,
        isMusclePrioritized: "true",
        changedValues: {isMusclePrioritized: "true"},
        setFormQuestionsList: jest.fn(),
    }
    wrapper = shallow(<ProfileSetupPage {...mockProps} />);
})

describe('<ProfileSetupPage />', () => {
  it("expect to render container without crashing with no props", () => {
    const mockProps = {
        currentUser: {},
        loading: true,
        getState: jest.fn(),
        subscribe: jest.fn(),
        dispatch: jest.fn(),
    }
    wrapper = shallow(
                <Provider store={mockProps}>
                    <ProfileSetupPageContainer {...mockProps} />
                </Provider>)
    expect(wrapper).toMatchSnapshot();
  })
  it("expect to render setup page on signup", () => {
    expect(wrapper).toMatchSnapshot();
  })
});