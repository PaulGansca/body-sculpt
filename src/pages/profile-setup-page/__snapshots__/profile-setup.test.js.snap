// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileSetupPage /> expect to render container without crashing with no props 1`] = `ShallowWrapper {}`;

exports[`<ProfileSetupPage /> expect to render setup page on signup 1`] = `ShallowWrapper {}`;
